python/lsst/ts/mtmount/mtmount_csc.py

```diff
@@ -1659,25 +1651,151 @@ async def do_stopTracking(self, data):
+                ),
+                do_lock=True,
+            )
+            await cmd_futures.done
+            # leaving the telescope with the park settings loaded
+            # because this is needed to unpark it.
 
-    async def _do_restoreDefaultSettings(self, data):
+    async def do_restoreDefaultSettings(self, data):
         self.assert_enabled()
 
-        raise salobj.ExpectedError("Command not implemented yet.")
+        async with self.in_progress_loop(
+            ack_in_progress=self.cmd_restoreDefaultSettings.ack_in_progress, data=data
+        ):
+            await self.handle_apply_settings_set(
+                restore_defaults=True,
+                settings_to_apply=[],
+            )
 
-    async def _do_unpark(self, data):
+    async def do_unpark(self, data):
         self.assert_enabled()
 
-        raise salobj.ExpectedError("Command not implemented yet.")
+        async with self.in_progress_loop(
+            ack_in_progress=self.cmd_park.ack_in_progress, data=data
+        ):
+            # move telescope to the unparked position
+
+            unpark_elevation = None
+            unpark_azimuth = None
+
+            async with salobj.Remote(
+                self.domain,
+                "MTMount",
+                include=["elevation", "azimuth"],
+                readonly=True,
+            ) as mtmount_remote:
+                current_elevation = await mtmount_remote.tel_elevation.next(
+                    flush=True, timeout=TELEMETRY_START_TIMEOUT
+                )
+                current_azimuth = await mtmount_remote.tel_azimuth.next(
+                    flush=True, timeout=TELEMETRY_START_TIMEOUT
+                )
+                unpark_elevation = (
+                    current_elevation.actualPosition
+                )  # this is going to be redacted furthermore
+                unpark_azimuth = current_azimuth.actualPosition
+
+            # Check it the telescope is pointing at horizon or zenith
+            if unpark_elevation >= self.evt_elevationControllerSettings.data.maxL1Limit:
+                self.log.info(
+                    f"Current telescope elevation at {unpark_elevation:.2f}. "
+                    "Unparking from Zenith."
+                )
+                unpark_elevation = (
+                    self.evt_elevationControllerSettings.data.maxL1Limit
+                    - self.limits_margin
+                )
+
+            elif (
+                unpark_elevation <= self.evt_elevationControllerSettings.data.minL1Limit
+            ):
+                self.log.info(
+                    f"Current telescope elevation at {unpark_elevation:.2f}. "
+                    "Unparking from Horizon."
+                )
+                unpark_elevation = (
+                    self.evt_elevationControllerSettings.data.minL1Limit
+                    + self.limits_margin
+                )
+            else:
+                raise RuntimeError(
```

Comment:
Would it be more consistent to refer to the valid park positions here like is done in line 1679?

Reply:
I don't think so. What this is doing is detecting if the telescope needs to be unparked at all. If the telescope is in the operational range (between the min/max elevation) there is no reason to unpark it. It will be already ready for operation. This error is just to prevent system from going through the whole process (which will take some time) unnecessarily.